tests/test_black.py

```diff
@@ -1296,6 +1296,28 @@ def test_read_pyproject_toml(self) -> None:
         self.assertEqual(config["exclude"], r"\.pyi?$")
         self.assertEqual(config["include"], r"\.py?$")
 
+    def test_black_replace_config(self) -> None:
+        test_toml_file = THIS_DIR / "test_replace.toml"
+        fake_ctx = FakeContext()
+        black.read_pyproject_toml(fake_ctx, FakeParameter(), str(test_toml_file))
+        config = fake_ctx.default_map
+        # `0` in `test_replace.toml` and `1` in `_black_config.toml`. Should be `1` as
```

Comment:
What is the reason for this behavior? The original use case was to make it possible to have a base config that other config files in a family of projects can import. For that use case, it makes more sense to me to have `test_replace.toml` win.

Either way, this behavior should be documented.

Reply:
Isn't this the required behaviour? The linked config would over-write the root config.

Reply:
Why? It doesn't make sense to me to allow other config in the toml that does `config = ` if we throw it away.

Reply:
A simple override could be better to not have weird issues and inconsistencies. And we could even warn or *error* on other values beyond `config`. But anyways, merging wasn't required at least in the original use case of multiple libraries sharing the same configuration in #1826. I'm still leaning on not merging. Not sure about it for CLI flags though, intuitively they should be prioritised.

Reply:
We discussed this in the discord channel and the commited solution was decided.